psqlextra/partitioning/plan.py

```diff
@@ -37,14 +38,26 @@ def apply(self, using: Optional[str]) -> None:
 
         connection = connections[using or "default"]
 
-        with transaction.atomic():
-            with connection.schema_editor() as schema_editor:
-                for partition in self.creations:
-                    partition.create(
-                        self.config.model,
-                        schema_editor,
-                        comment=AUTO_PARTITIONED_COMMENT,
-                    )
+        if self.creations:
+            with transaction.atomic():
+                with connection.schema_editor() as schema_editor:
+                    for partition in self.creations:
+                        partition.create(
+                            self.config.model,
+                            schema_editor,
+                            comment=AUTO_PARTITIONED_COMMENT,
+                        )
+
+        if self.deferred_creations:
+            with transaction.atomic():
+                with connection.schema_editor() as schema_editor:
+                    for partition in self.creations:
+                        partition.create(
+                            self.config.model,
+                            schema_editor,
+                            comment=AUTO_PARTITIONED_COMMENT,
+                            defer_attach=True,
```

Comment:
Won't this crash on other partitioning methods like list or hash?

Reply:
Yes, it will. I thought we'll just support range partition (and its derivatives).

Reply:
To make the whole thing more legit, I could implement the same code for hash and list partitions, or throw "not implemented" exception when defer_attach is passed.